### ISSUE TITLE
Syntax error

### ISSUE THREAD

**Issue by ramzcode** [bartmichu/node-urbackup-server-api#3]
Hi I am new to JS and API stuff. But tried to run the script like node scriptname.js but i am getting syntax errors.

Not sure if i am missing something.

node urbackup-API6.js
#clearLoginStatus () {
                    ^

SyntaxError: Unexpected token '('
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/root/urbackup-API6.js:1:28)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)

**Comment by bartmichu** [comment]
Hi! What version of Node are you using (node --version)? It would be easier if you provided some code.

Example CLI application could look like this:

```js
const { program } = require('commander');
const { UrbackupServer } = require('urbackup-server-api');

process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
process.removeAllListeners('warning');

const server = new UrbackupServer({ url: 'https://127.0.0.1:443', username: 'admin', password: 'whatever' });

program
  .command('raw-status')
  .description('Get raw response of status API call')
  .action(() => {
    statusCall('raw-status');
  });

program
  .command('all-clients')
  .description('Get all clients')
  .option('--number', 'get total number instead of array with client names', false)
  .action((options) => {
    statusCall('all-clients', options);
  });

program
  .command('ok-clients')
  .description('Get OK clients')
  .option('--skip-file', 'skip file backups', false)
  .option('--skip-image', 'skip image backups', false)
  .option('--number', 'get total number instead of array with client names', false)
  .action((options) => {
    statusCall('ok-clients', options);
  });

program
  .command('failed-clients')
  .description('Get failed clients (clients with failed backup or without a recent backup as configured in UrBackup Server)')
  .option('--skip-file', 'skip file backups', false)
  .option('--skip-image', 'skip image backups', false)
  .option('--skip-blank', 'skip blank clients', false)
  .option('--number', 'get total number instead of array with client names', false)
  .action((options) => {
    statusCall('failed-clients', options);
  });

program
  .command('online-clients')
  .description('Get online clients')
  .option('--skip-blank', 'skip blank clients', false)
  .option('--number', 'get total number instead of array with client names', false)
  .action((options) => {
    statusCall('online-clients', options);
  });

program
  .command('offline-clients')
  .description('Get offline clients')
  .option('--skip-blank', 'skip blank clients', false)
  .option('--number', 'get total number instead of array with client names', false)
  .action((options) => {
    statusCall('offline-clients', options);
  });

async function statusCall (command, options) {
  const currentTimestamp = Math.round(new Date().getTime() / 1000);

  try {
    await server.getStatus({ includeRemoved: false }).then(statusResponse => {
      const matchingClients = [];

      switch (command) {
        case 'raw-status':
          console.log(statusResponse);
          return statusResponse;
 
        case 'all-clients':
          for (const clientStatus of statusResponse) {
            matchingClients.push(clientStatus.name);
          }
          break;

        case 'ok-clients':
          for (const clientStatus of statusResponse) {
            if (options.skipFile !== true) {
              if (clientStatus.file_disabled !== true && clientStatus.file_ok === true) {
                matchingClients.push(clientStatus.name);
                continue;
              }
            }

            if (options.skipImage !== true) {
              if (clientStatus.image_disabled !== true && clientStatus.image_ok === true) {
                matchingClients.push(clientStatus.name);
                continue;
              }
            }
          }
          break;

        case 'failed-clients':
          for (const clientStatus of statusResponse) {
            if (options.skipFile !== true) {
              if ((options.skipBlank !== true || (options.skipBlank === true && clientStatus.lastbackup !== 0)) && clientStatus.file_disabled !== true && clientStatus.file_ok !== true) {
                matchingClients.push(clientStatus.name);
                continue;
              }
            }

            if (options.skipImage !== true) {
              if ((options.skipBlank !== true || (options.skipBlank === true && clientStatus.lastbackup_image !== 0)) && clientStatus.image_disabled !== true && clientStatus.image_ok !== true) {
                matchingClients.push(clientStatus.name);
                continue;
              }
            }
          }
          break;

        case 'online-clients':
        case 'offline-clients':
          for (const clientStatus of statusResponse) {
            if ((options.skipBlank !== true || (options.skipBlank === true && clientStatus.lastbackup !== 0)) && (clientStatus.online === (command === 'online-clients'))) {
              matchingClients.push(clientStatus.name);
            }
          }
          break;
      }

      console.log(options.number === true ? matchingClients.length : matchingClients);
      return options.number === true ? matchingClients.length : matchingClients;
    });
  } catch (e) {
    console.debug(e.message);
  }
}

(async () => {
  await program.parseAsync(process.argv);
})().catch(e => {
  // Deal with the fact the chain failed
});
```

For more examples please look into my other project, urbstat, which lives at https://github.com/bartmichu/urbstat.

It is also JavaScript, but not Node. It turned out that Deno is a much better fit for my project so I've ported this module to Deno (it was almost copy+paste) and that's where I'm putting all the effort at the moment.

**Comment by ramzcode** [comment]
My node version is v12.22.9, using Ubuntu 22.04.

const { UrbackupServer } = require('urbackup-server-api');

// When troubleshooting TSL connections with self-signed certificates you may try to disable certificate validation. Keep in mind that it's strongly discouraged for production use.
//process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

const server = new UrbackupServer({ url: 'http://127.0.0.1:55414', username: 'admin', password: 'admin' });

(async () => {
  try {
    const allClients = await server.getStatus();
    console.log('Clients with failed file backups:');
    allClients.filter(client => client.file_ok === false).forEach(client => console.log(client.name));
  } catch (error) {
    // Deal with it
  }
})();

**Comment by bartmichu** [comment]
> My node version is v12.22.9, using Ubuntu 22.04

Oh dear, that explains it. You are using an outdated version of Node that has been without security support for over a year now. I understand that this is the version available in the 22.04 repositories, but even the official website recommends the excellent [NodeSource repository for Ubuntu](https://github.com/nodesource/distributions#using-ubuntu-2). The LTS (Long-Term Support) version of Node is your best choice, which is version 18 currently. Installing it is really simple. Purge your v12 and then:

```shell
curl -fsSL https://deb.nodesource.com/setup_18.x | sudo -E bash - &&\
sudo apt-get install -y nodejs
```

Please avoid starting your JavaScript journey with such an outdated version of Node :) Just to be sure I did run your example - it works fine with current Node. Thanks for pointing that out, I will update the documentation.

**Comment by ramzcode** [comment]
Great !! Thanks a lot now understood the reason. 
Once again thank you for guiding and now it works.
